### PR TITLE
Fix timer os metric build formula watch

### PR DESCRIPTION
**Pull request by brunats**
**- What I did**
Add stop condition to the hit build formula --watch command. Now the process can be completed correctly when the user presses 'Ctrl + c'.

This corrects the timing of sending metrics that is next to the process stop.

>stop

**- How to verify it**

**- Description for the changelog**

Fix timer os metric build formula watch

### CODE REVIEW

File: pkg/formula/watcher/watcher.go [hunk 1]
```diff
@@ -78,11 +97,9 @@ func (w *WatchManager) Watch(workspacePath, formulaPath string) {
 
 	w.build(workspacePath, formulaPath)
 
-	watchText := fmt.Sprintf("Watching dir %s \n", formulaPath)
-	prompt.Info(watchText)
-
-	startTime := time.Now().Second()
-	w.sendMetric(float64(startTime))
```
Comment (henriquemoraeszup) [inline pkg/formula/watcher/watcher.go:85]: just curious, where is the start time being collected now? And how does it differ a start from an end? `sendMetric` interprets it on the 2nd call?

Comment (brunasilvazup) [reply pkg/formula/watcher/watcher.go:85]: Hi, I also had this question, I just changed the location, but I didn't check the difference.

In the main it uses a function to calculate the difference.
And checking now I saw that leaving the send metric here he sends twice.

In the case of the main send metric it uses save the start time and get the time again at the end of the execution.

File: pkg/formula/watcher/watcher.go [hunk 2]
```diff
@@ -54,8 +57,22 @@ func New(
 	}
 }
 
+func (w *WatchManager) closeWatch() {
+	currentTime := time.Now().Second()
+	w.sendMetric(float64(currentTime))
```
Comment (henriquemoraeszup) [inline pkg/formula/watcher/watcher.go:62]: is this asynchronous or should the `stopping` text come before it?

**Review by henriquemoraeszup — APPROVED** [review]
Just one comment but lgtm

File: pkg/formula/watcher/watcher.go (hunk 2 above)
Comment (brunasilvazup) [reply pkg/formula/watcher/watcher.go]: sendMetric is asynchronous (:

**Review by victorschumacherzup — APPROVED** [review]

**Comment by brunasilvazup** [comment]
\merge qa

**Review by lucasdittrichzup — APPROVED** [review]

**Pull request merged**